### PR TITLE
Refactor SCAI generator APIs into pkg/

### PR DESCRIPTION
**Pull request by marcelamelara**
This PR refactors the core SCAI predicate generator APIs into `pkg/` for external usage outside the CLI. 

An anticipated usage of these refactored APIs will be in the [build image attestation PoC](https://github.com/chkimes/image-attestation) for the SLSA workstream on [Hardware Attested Platforms](https://github.com/slsa-framework/slsa/issues/975).

This also fixes a couple of related documentation bugs.

### CODE REVIEW

File: scai-gen/pkg/generators/v1.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"os"
+	"strings"
+
+	"github.com/in-toto/scai-demos/scai-gen/pkg/policy"
+
+	ita "github.com/in-toto/attestation/go/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func NewRdForFile(filename string, name string, uri string, hashAlg string, withContent bool, mediaType string, downloadLocation string, annotations *structpb.Struct) (*ita.ResourceDescriptor, error) {
```
Comment (adityasaky) [inline scai-gen/pkg/generators/v1.go:15]: nit: docstring

Comment (adityasaky) [reply scai-gen/pkg/generators/v1.go:15]: also this is a long line, a small-delta way to cut this down would be to not repeat the parameter types for consecutive args of the same type.

```
func NewRdForFile(filename, name, uri, hashAlg string, withContent bool, mediaType, downloadLocation string, annotations *structpb.Struct) (*ita.ResourceDescriptor, error) {
```

If `withContent` is moved, you could get away with using a single `string` for all the params...

File: scai-gen/pkg/generators/v1.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	err = rd.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("invalid resource descriptor: %w", err)
+	}
+
+	return rd, nil
+}
+
+func NewRdForRemote(name string, uri string, hashAlg string, digest string, downloadLocation string, annotations *structpb.Struct) (*ita.ResourceDescriptor, error) {
```
Comment (adityasaky) [inline scai-gen/pkg/generators/v1.go:58]: similar thing here

File: scai-gen/pkg/generators/v1.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	digestSet := make(map[string]string)
+	if len(digest) > 0 {
+		// the in-toto spec expects a hex-encoded string in DigestSets
+		// https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md
+		_, err := hex.DecodeString(digest)
```
Comment (adityasaky) [inline scai-gen/pkg/generators/v1.go:63]: perhaps this should happen in in-toto/attestation in .Validate().

**Review by adityasaky — COMMENTED** [review]
Various comments, I think this is basically fine. Is `pkg/` just to adopt the convention?

File: scai-gen/pkg/generators/v1.go (hunk 3 above)
Comment (marcelamelara) [reply scai-gen/pkg/generators/v1.go:63]: good point. I'll open a PR for this.

File: scai-gen/pkg/generators/v1.go (hunk 1 above)
Comment (marcelamelara) [reply scai-gen/pkg/generators/v1.go:15]: Ah yes, sorry about missing the docstrings. Will add. Also forgot I could group params of the same type, thanks!

File: scai-gen/pkg/generators/v1.go (hunk 3 above)
Comment (marcelamelara) [reply scai-gen/pkg/generators/v1.go:63]: done in https://github.com/in-toto/attestation/pull/330

**Comment by SantiagoTorres** [comment]
@alanssitis ptal!

**Review by alanssitis — APPROVED** [review]
lgtm, I wonder if we should consider moving `scai-gen` out of the demo?

I'll create an issue discussing this more in-depth in a bit :P

**Comment by marcelamelara** [comment]
Should we add @alanssitis as a maintainer?

Also, totally agree that scai-gen should eventually be moved out of this repo, and probably into go-witness as part of the broader in-toto golang consolidation efforts.

**Pull request merged**